doc/en/source/system_overview.rst

```diff
@@ -14,6 +14,86 @@ illustrated in below figure. All integrated features that are running in FogFlow
 .. figure:: figures/FogFlow_System_Design.png
 
 
+
+FogFlow is now supporting graph database to store internal entity data for later use. In FogFlow system, the interaction point of graph database is designer.
+So FogFlow entity that create either by FogFlow web browser or via any other client such as curl or python client,
+the entity creation request directly goes to Designer. Then Designer will send this entity request on cloud broker to 
+register the entity and at the same time Designer will send this data on DGraph to store it into database.
+
+Whenever FogFlow system will restart, Designer will trigger a query request to DGraph and get all the stored entities
+from DGraph then send these entities to cloud broker to register them.
+
+.. note:: FogFlow Designer is dependent on Discovery and Cloud Broker.
+
+The integration of FogFlow Designer and DGraph is done via gRPC, where a Dgraph client is implemented with Designer server as in below code
+using gRPC. Default port of gRPC is 9080, in FogFlow 9082 port is used.
+
+.. code-block:: console
+
+   /*
+   creating grpc client for making connection with dgraph
+   */
+   function newClientStub() {
+       return new dgraph.DgraphClientStub(config.HostIp+":"+config.grpcPort, grpc.credentials.createInsecure());
+   }
+
+   // Create a client.
+   function newClient(clientStub) {
+      return new dgraph.DgraphClient(clientStub);
+   }
+   
+   
+Whenever any new FogFlow entity creates ( via web browser or via other clients) the request goes to designer, then designer 
```

Comment:
[Trivial] Please correct the sentence.
 Whenever new FogFlow entity gets created by web browser  the request goes to designer or user directly creates entity on designer using any client. Then designer perform two tasks